covid_api/db/static/datasets/__init__.py

```diff
@@ -16,242 +23,183 @@ class DatasetManager(object):
+                )
+
     def get(self, spotlight_id: str, api_url: str) -> Datasets:
         """
         Fetches all the datasets avilable for a given spotlight. If the
         spotlight_id provided is "global" then this method will return
         all datasets that are NOT spotlight specific. Raises an
         `InvalidIdentifier` exception if the provided spotlight_id does
         not exist.
+
+        Params:
+        -------
+        spotlight_id (str): spotlight id to return datasets for
+        api_url(str): {scheme}://{host} of request originator in order
+            to return correctly formated source urls
+
+        Returns:
+        -------
+        (Datasets) pydantic model contains a list of datasets' metadata
         """
-        global_datasets = self._get_global_datasets()
-        global_datasets = self._overload_domain(datasets=global_datasets)
+
+        global_datasets = self._process(
+            self._load_domain_metadata()["global"],
+            api_url=api_url,
+            spotlight_id="global",
+        )
 
         if spotlight_id == "global":
-            return self._prep_output(global_datasets, api_url=api_url)
+            return Datasets(datasets=[dataset.dict() for dataset in global_datasets])
 
         # Verify that the requested spotlight exists
         try:
             site = sites.get(spotlight_id)
         except InvalidIdentifier:
             raise
 
-        # Append "EUPorts" to the spotlight ID's if the requested spotlight id
-        # was one of "du" or "gh", since certain datasets group both spotlights
-        # under a single value: "EUPorts". It's then necessary to search,
-        # and extract domain for each option ("du"/"gh" and "EUPorts") separately
-
-        spotlight_ids = [site.id]
-        if site.id in ["du", "gh"]:
-            spotlight_ids.append("EUPorts")
-
-        spotlight_datasets = {}
-
-        for spotlight_id in spotlight_ids:
-            # find all "folders" in S3 containing keys for the given spotlight
-            # each "folder" corresponds to a dataset.
-            spotlight_dataset_folders = get_dataset_folders_by_spotlight(
-                spotlight_id=spotlight_id
-            )
-            # filter the dataset items by those corresponding the folders above
-            # and add the datasets to the previously filtered `global` datasets
-            datasets = self._filter_datasets_by_folders(
-                folders=spotlight_dataset_folders
-            )
-
-            datasets = self._overload_spotlight_id(
-                datasets=datasets, spotlight_id=spotlight_id
-            )
-
-            datasets = self._overload_domain(
-                datasets=datasets, spotlight_id=spotlight_id
-            )
-            spotlight_datasets.update(datasets)
+        spotlight_datasets = self._process(
+            self._load_domain_metadata()[site.id],
+            api_url=api_url,
+            spotlight_id=site.id,
```

Comment:
The `api_url` and `spotlight_id` parameters are use to insert the correct base API and spotlight into the returned dataset metadata objects